src/components/ProductInfo/ProductInfo.component.jsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useContext, useEffect } from "react";
+import CartContext from "../../state/CartContext";
+
+function ProductInfo(props) {
+  const { state, dispatch } = useContext(CartContext);
+  const [availableStock, setAvailableStock] = useState([1]);
+  const [selectedQuantity, setSelectedQuantity] = useState(1);
+
+  function handleSelect(e) {
+    setSelectedQuantity(parseInt(e.target.value));
+  }
+
+  function handleAddToCart() {
+    if (selectedQuantity > availableStock.length) {
+      alert("Not enough stock");
+      updateAvailableStock();
+    } else {
+      dispatch({
+        type: "ADD_TO_CART",
+        payload: {
+          id: props.id,
+          name: props.data.name,
+          image: props.data.mainimage.url,
+          price: props.data.price,
+          quantity: selectedQuantity,
+        },
+      });
+      updateAvailableStock();
+    }
+  }
+
+  // Function to check how many of the same product are in the cart
+  function checkCartQty() {
+    let cart = state.cart;
+    let cartQty = 0;
+    if (cart) {
+      cart.forEach((cartItem) => {
+        if (cartItem.id === props.id) {
+          cartQty = parseInt(cartItem.quantity);
+        }
+      });
+      return cartQty;
+    }
+  }
+
+  function updateAvailableStock() {
+    let stock = props.data.stock;
+    let cartQty = checkCartQty();
+    let availableStock = Array.from({ length: stock }, (v, k) => k + 1);
+    availableStock = availableStock.slice(0, availableStock.length - cartQty);
+    setAvailableStock(availableStock);
+  }
+
+  useEffect(() => {
+    updateAvailableStock();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [state.cart]);
```

Comment:
```suggestion
  }, [state.cart, updateAvailableStock]);
```
this should help you remove the eslint rule

Reply:
Yup, I refactored the functions using useCallback